### PR TITLE
chore: update tests to run on node 20

### PR DESCRIPTION
**Pull request by 0xDEC0DE**
Fix workflows, tox.ini

Drop Python < 3.8 support

### CODE REVIEW

**Comment by bigjools** [comment]
Merge this with the other branch

File: .github/workflows/tests.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10", "pypy3"]
+        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12"]
```
Comment (bigjools) [inline .github/workflows/tests.yml:12]: Why remove pypy?

Comment (0xDEC0DE) [reply .github/workflows/tests.yml:12]: The previous tests claimed it was 3.6, i.e., it no longer exists.

File: .github/workflows/tests.yml [hunk 2]
```diff
@@ -9,7 +9,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        python-version: ["3.6", "3.7", "3.8", "3.9", "3.10", "pypy3"]
+        python-version: ["3.8", "3.9", "3.10"]
```
Comment (bigjools) [inline .github/workflows/tests.yml:12]: You need pypy3.10 it seems.

**Review by bigjools — APPROVED** [review]

File: .github/workflows/tests.yml (hunk 2 above)
Comment (bigjools) [reply .github/workflows/tests.yml:12]: Ok, in the next branch which fixes the pypy error ...

**Pull request merged**